core/sysparam.c

```diff
@@ -33,11 +31,14 @@
  */
 #define SCAN_BUFFER_SIZE 8 // words
 
-/* The size of the temporary buffer used for reading back and verifying data
- * written to flash.  Making this larger will make the write-and-verify
- * operation slightly faster, but will use more heap during writes
+/* The size in words of the buffer used for reading keys when searching for a
+ * match, for reading payloads to check if the value has changed, and reading
+ * back from the flash to verify writes. Will work well if big enough for
+ * commonly used keys, and must be at least one word. Stack allocated so not too
+ * large!
```

Comment:
Actually, I added static during debugging and just forgot to remove it... I'm not sure what is actually better for that buffer to be static or be in stack.
I would actually prefer it to be in stack as it was before. As there's another such buffer in function `_compare_payload` which is in stack and it's better to be consistent.